src/Api.Rest.Tests/HttpClient/Health/HealthServiceRestClientTest.cs

```diff
@@ -0,0 +1,140 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2021                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Api.Rest.Tests.HttpClient.Health
+{
+	#region usings
+
+	using System;
+	using System.Net;
+	using System.Threading.Tasks;
+	using FluentAssertions;
+	using NUnit.Framework;
+	using Zeiss.PiWeb.Api.Rest.Contracts;
+	using Zeiss.PiWeb.Api.Rest.HttpClient.Health;
+
+	#endregion
+
+	[TestFixture]
+	public class HealthServiceRestClientTest
+	{
+		#region constants
+
+		private const int Port = 8081;
+
+		#endregion
+
+		#region members
+
+		private static readonly Uri Uri = new ( $"http://localhost:{Port}/" );
+
+		#endregion
+
+		#region methods
+
+		[Test]
+		public async Task GetReadiness_ApiResponseIs200Healthy_ReturnsHealthy()
+		{
+			using var server = WebServer.StartNew( Port );
+			using var client = new HealthServiceRestClient( Uri );
+			server.RegisterDefaultResponse( "Healthy" );
+
+			var result = await client.GetReadiness();
+
+			result.Should().Be( HealthCheckResultType.Healthy );
+		}
+
+		[Test]
+		public async Task GetReadiness_ApiResponseIs200Degraded_ReturnsDegraded()
+		{
+			using var server = WebServer.StartNew( Port );
+			using var client = new HealthServiceRestClient( Uri );
+			server.RegisterDefaultResponse( "Degraded" );
+
+			var result = await client.GetReadiness();
+
+			result.Should().Be( HealthCheckResultType.Degraded );
+		}
+
+		[Test]
+		public async Task GetReadiness_ApiResponseIs503_ReturnsUnhealthy()
+		{
+			using var server = WebServer.StartNew( Port );
+			using var client = new HealthServiceRestClient( Uri );
+			server.RegisterDefaultResponse( "Unhealthy", HttpStatusCode.ServiceUnavailable );
+
+			var result = await client.GetReadiness();
+
+			result.Should().Be( HealthCheckResultType.Unhealthy );
+		}
+
+		[Test]
+		public async Task GetReadiness_ApiResponseIs500_ThrowsException()
```

Comment:
I am still thinking about this, would it make more sense to treat a complete request failure as an unhealthy state? Or at least for 500. Maybe we should also treat 404 as "Server does not yet have this endpoint"?

Reply:
No i would not do that. Because network issues between health check client <--> server can also occur. And those issues should not be interpreted as "PiWeb Server is unhealthy".

Regarding the 404 thing... My suggestion would be that this should be interpreted by the clients who are using the API SDK. Not by the SDK himself.

Reply:
I agree with the network issues, so this could stay the way it is.
Just for the check, if the Server actually supports the feature: we already do such things in the SDK, so that a user must not know about "since when is this feature avaiable?", e.g. with our FeatureMatrix and version checks. The user just calls the SDK method, and gets an exception, e.g. that the server is too old.

Reply:
GetInterfaceVersion implemented